DANS-KNAW/easy-update-metadata-dataset
# EASY-1003 update dataset metadata

fixes EASY-1003 and future similar issues
#### When applied it will

change specified metadata of datasets
#### Where should the reviewer @DANS-KNAW/easy start?

Everything is new, so perhaps start with the readme and CommandSpec
#### How should this be manually tested?
- deploy on deasy as described in related PR or as described the instructions at the bottom of the readme, deasy should have a clean version of the preloaded dataset titled `Groningen in Oorlogstijd, interview 01` **note** fix the id in the links below
- copy  `src/test/resources/deasy-data.csv` to `easy-dtap/shared` 
- on deasy `sudo /home/easy-tool/easy-update-metadata-dataset/bin/run.sh -s EMD -t accessRights  /vagrant/shared/deasy-data.csv` it will ask for fedora/ldap credentials, though you can also specify them as options; you can find the values with `egrep '(fedora.admin|ldap.sec)' /opt/easy-webui/cfg/application.properties`
- same command with `-s DC -t rights`
- to apply the changes for real use the options `-s EMD -t accessRights --doUpdate`
- use fedora to check the new rights http://deasy.dans.knaw.nl:8080/fedora/objects/easy-dataset:9999/datastreams/EMD/content (or replace EMD with DC or whatever is applicable)
- in case of EMD also check http://deasy.dans.knaw.nl/ui/datasets/id/easy-dataset:9999/tab/1
#### related pull requests on github

| repo | PR | note |
| --- | --- | --- |
| easy-dtap | TODO | depends on this PR |
| [easy-module-archetype](https://github.com/DANS-KNAW/easy-module-archetype) | n.a. | should apply the changes up to e7cac88 |
| easy-app | [#17](https://github.com/DANS-KNAW/easy-app/pull/17) (syntegra) | provides input for this command, updates <strike>DC and</strike> file rights |

https://github.com/DANS-KNAW/easy-dtap/pull/20
